inlong-agent/agent-core/src/main/java/org/apache/inlong/agent/core/AgentStatusManager.java

```diff
@@ -0,0 +1,279 @@
+import static org.apache.inlong.agent.constant.FetcherConstants.AGENT_GLOBAL_WRITER_PERMIT;
+import static org.apache.inlong.agent.constant.FetcherConstants.AGENT_MANAGER_ADDR;
+import static org.apache.inlong.agent.constant.FetcherConstants.AGENT_MANAGER_AUTH_SECRET_ID;
+import static org.apache.inlong.agent.constant.FetcherConstants.AGENT_MANAGER_AUTH_SECRET_KEY;
+
+/**
+ * Collect various indicators of agent processes for backend problem analysis
+ */
+public class AgentStatusManager {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AgentStatusManager.class);
+    public static final String INLONG_AGENT_SYSTEM = "inlong_agent_system";
+    public static final String INLONG_AGENT_STATUS = "inlong_agent_status";
+
+    private static AgentStatusManager manager = null;
+    private final AgentConfiguration conf;
+    private final SimpleDateFormat format = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss"); // 设置格式
+    private Runtime runtime = Runtime.getRuntime();
+    final long GB = 1024 * 1024 * 1024;
+    private OperatingSystemMXBean osMxBean;
+    private ThreadMXBean threadBean;
+    private long INVALID_CPU = -1;
+    private RuntimeMXBean runtimeMXBean = ManagementFactory.getRuntimeMXBean();
+    private ThreadMXBean threadMXBean = ManagementFactory.getThreadMXBean();
+    private AgentManager agentManager;
+    public static AtomicLong sendDataLen = new AtomicLong();
+    public static AtomicLong sendPackageCount = new AtomicLong();
+    private DefaultMessageSender sender;
+    private List<String> statusFieldsPre = Lists.newArrayList();
+    private String processStartupTime = format.format(runtimeMXBean.getStartTime());
+    private String systemStartupTime = ExcuteLinux.exeCmd("who -b|awk '{print $(NF-1), $NF}'").replaceAll("\r|\n", "");
+
+    private AgentStatusManager(AgentManager agentManager) {
+        this.agentManager = agentManager;
+        this.conf = AgentConfiguration.getAgentConf();
+        osMxBean = ManagementFactory.getOperatingSystemMXBean();
+        threadBean = ManagementFactory.getThreadMXBean();
+        initStatusFieldsPre();
+        createMessageSender();
+    }
+
+    public static AgentStatusManager getInstance(AgentManager agentManager) {
+        if (manager == null) {
+            synchronized (AgentStatusManager.class) {
+                if (manager == null) {
+                    manager = new AgentStatusManager(agentManager);
+                }
+            }
+        }
+        return manager;
+    }
+
+    public static AgentStatusManager getInstance() {
+        if (manager == null) {
+            throw new RuntimeException("HeartbeatManager has not been initialized by agentManager");
+        }
+        return manager;
+    }
+
+    public void sendStatusMsg(List<String> fields) {
+        if (sender == null) {
+            LOGGER.error("sender is null");
+            createMessageSender();
+            return;
+        }
+        SendResult ret = sender.sendMessage(
```

Comment:
The sender may have failed to be created previously, causing NPE. It is recommended to make a null judgment on the sender again.

Reply:
there is an return